libkbfs/favorites.go

```diff
@@ -107,6 +112,87 @@ func NewFavorites(config Config) *Favorites {
 	return newFavoritesWithChan(config, make(chan *favReq, 100))
 }
 
+func (f *Favorites) readCacheFromDisk(ctx context.Context) error {
+	// Read the encrypted cache from disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// Send the data to the service to be decrypted
+	decryptedData, err := f.config.KeybaseService().DecryptFavorites(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	// Decode the data into the a map
+	decoder := gob.NewDecoder(bytes.NewBuffer(decryptedData))
```

Comment:
Why use gob?  As far as I know this is the only place we use this.

I think a more standard way would be to define a struct that holds the cache map, and encode that using `f.config.Codec()` (which would use msgpack). Besides being more standard, that would let us add other fields in the future.

Also it might be good to add a version number to the struct holding the map as well, in case the format (or interpretation of the format) of the cache needs to change in the future.  And after decrypting, we check that the version number matches what we expect or else we panic.